redisson/src/main/java/org/redisson/RedissonPermitExpirableSemaphore.java

```diff
@@ -295,114 +300,147 @@ private CompletableFuture<String> acquireAsync(int permits, RedissonLockEntry en
+            RFuture<List<String>> future = tryAcquireAsync(ids, timeoutDate);
+            return commandExecutor.handleNoSync(future, () -> releaseAsync(ids));
         });
     }
 
-    private RFuture<String> tryAcquireAsync(byte[] id, int permits, long timeoutDate) {
-        return commandExecutor.syncedEval(getRawName(), ByteArrayCodec.INSTANCE, RedisCommands.EVAL_PERMIT_DATA,
-                  "local expiredIds = redis.call('zrangebyscore', KEYS[2], 0, ARGV[4], 'limit', 0, ARGV[1]); " +
-                  "if #expiredIds > 0 then " +
-                      "redis.call('zrem', KEYS[2], unpack(expiredIds)); " +
-                      "local value = redis.call('incrby', KEYS[1], #expiredIds); " +
-                      "if tonumber(value) > 0 then " +
-                          "redis.call(ARGV[6], KEYS[3], value); " +
-                      "end;" +
-                  "end; " +
-                  "local value = redis.call('get', KEYS[1]); " +
-                  "if (value ~= false and tonumber(value) >= tonumber(ARGV[1])) then " +
-                      "redis.call('decrby', KEYS[1], ARGV[1]); " +
-                      "redis.call('zadd', KEYS[2], ARGV[2], ARGV[3]); " +
-
-                      "local ttl = redis.call('pttl', KEYS[1]); " +
-                      "if ttl > 0 then " +
-                          "redis.call('pexpire', KEYS[2], ttl); " +
-                      "end; " +
-
-                      "return ARGV[3]; " +
-                  "end; " +
-                  "local v = redis.call('zrange', KEYS[2], 0, 0, 'WITHSCORES'); " +
-                  "if v[1] ~= nil and v[2] ~= ARGV[5] then " +
-                      "return ':' .. tostring(v[2]); " +
-                  "end " +
-                  "return nil;",
-                  Arrays.asList(getRawName(), timeoutName, channelName),
-                permits, timeoutDate, id, System.currentTimeMillis(), nonExpirableTimeout, getSubscribeService().getPublishCommand());
+    private RFuture<List<String>> tryAcquireAsync(List<String> ids, long timeoutDate) {
+        CompletionStage<List<String>> future = commandExecutor.syncedEval(getRawName(), ByteArrayCodec.INSTANCE, RedisCommands.EVAL_STRING,
+            "local expiredIds = redis.call('zrangebyscore', KEYS[2], 0, ARGV[4], 'limit', 0, ARGV[1]); " +
+                "if #expiredIds > 0 then " +
+                    "redis.call('zrem', KEYS[2], unpack(expiredIds)); " +
+                    "local value = redis.call('incrby', KEYS[1], #expiredIds); " +
+                    "if tonumber(value) > 0 then " +
+                        "redis.call(ARGV[6], KEYS[3], value); " +
+                    "end; " +
+                "end; " +
+                "local value = redis.call('get', KEYS[1]); " +
+                "if (value ~= false and tonumber(value) >= tonumber(ARGV[1])) then " +
+                    "redis.call('decrby', KEYS[1], ARGV[1]); " +
+                    "for key in string.gmatch(ARGV[3], '%w+') do " +
```

Comment:
why do we need this?

Reply:
I mean "string.gmatch". we need to avoid it

Reply:
Oh ok. The reason I used it that argument passed to script (List in our case) is converted to string. And I didn't find better way to unpack it back to list of values

Reply:
Here is the example how to pass collection as arguments org.redisson.RedissonTimeSeries#addAllAsync(java.util.Map<java.lang.Long,V>, long, java.util.concurrent.TimeUnit)

Reply:
Thanks! It helped.
I've changed it back to use byte arrays as ids and deleted "string.gmatch" usages